nntrainer/app_context.cpp

```diff
@@ -196,4 +303,31 @@ int AppContext::registerLayerPlugin(const std::string &path,
   return registerFactory<ml::train::Layer>(factory_func, type);
 }
 
+std::vector<int>
+AppContext::registerLayerFromDirectory(const std::string &base_path) {
+  DIR *dir = opendir(base_path.c_str());
+
+  NNTR_THROW_IF(dir == nullptr, std::invalid_argument)
+    << func_tag << "failed to open the directory: " << base_path;
+
+  struct dirent *entry;
+
+  std::vector<int> keys;
+  while ((entry = readdir(dir)) != NULL) {
+    if (endswith(entry->d_name, layerlib_suffix)) {
+      try {
```

Comment:
so.. to register custome layer, it must ends with ".......layer.so"??

Reply:
Yes. This is tentative. Any suggestions are welcome :)

Reply:
Is a suffix needed? IMO different folders for different types - layers, optimizers, etc. - might be better.